scalding-core/src/main/scala/com/twitter/scalding/typed/CoGrouped.scala

```diff
@@ -333,6 +333,8 @@ abstract class CoGroupedJoiner[K](inputSize: Int,
 
   override def getIterator(jc: JoinerClosure) = {
     val iters = (0 until distinctSize).map { jc.getIterator(_).asScala.buffered }
+    // Don't pattern match in performance-critical code
```

Comment:
that's not perf critical as much as the invariant that one of the iters has an item is not communicated in the types by cascading